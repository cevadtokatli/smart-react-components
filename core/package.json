{
  "name": "@smart-react-components/core",
  "version": "2.0.0-beta-0",
  "license": "MIT",
  "main": "index.js",
  "description": "SRC core includes the theme, Styled properties, and Element components.",
  "keywords": [
    "javascript",
    "typescript",
    "react",
    "styled-components"
  ],
  "author": {
    "name": "Cevad Tokatlı",
    "email": "cevadtokatli@hotmail.com"
  },
  "contributors": [
    {
      "name": "Cemil Tokatlı",
      "email": "cemil@tokatli.me"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/cevadtokatli/smart-react-components/tree/v2/core"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "yarn clear && tsc --outDir lib && yarn copy",
    "clear": "rm -rf lib",
    "copy": "cp package.json lib/package.json && cp README.md lib/README.md && cp ../LICENSE lib/LICENSE",
    "dev": "tsc -w --outDir lib",
    "lint": "eslint src",
    "release": "../scripts/release.sh core",
    "test": "jest"
  },
  "devDependencies": {
    "@types/color": "3.0.3",
    "@types/enzyme": "3.10.12",
    "@types/jest": "29.0.1",
    "@types/react": "18.0.19",
    "@types/react-dom": "18.0.6",
    "@types/styled-components": "5.1.26",
    "@wojtekmaj/enzyme-adapter-react-17": "0.6.7",
    "enzyme": "3.11.0",
    "eslint": "8.0.1",
    "jest": "29.0.3",
    "jsdom": "20.0.0",
    "jsdom-global": "3.0.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "styled-components": "5.3.5",
    "ts-jest": "29.0.0",
    "typescript": "4.8.3"
  },
  "dependencies": {
    "color": "4.2.3"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "styled-components": "^5.3.5"
  }
}
