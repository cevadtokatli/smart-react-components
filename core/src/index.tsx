export * from './theme'
export { default as createTheme } from './theme'
export * from './styled-props'
export * from './Element'
