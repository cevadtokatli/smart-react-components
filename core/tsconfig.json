{
  "extends": "../tsconfig.json",
  "include": ["src/**/*.tsx"],
  "exclude": ["node_modules", "src/**/*.test.tsx"]
}
