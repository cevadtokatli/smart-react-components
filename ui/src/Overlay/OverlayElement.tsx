import Div from '@smart-react-components/core/Element/Div'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { Breakpoint } from '@smart-react-components/core/types/grid'
import styled from 'styled-components'

const overlayCSS = ({ theme, hasBackground }) => `
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  height: 100%;
  left: 0;
  padding: ${theme.$.length.overlay.space};
  top: 0;
  touch-action: none;
  width: 100%;

  ${hasBackground
    ? `
      background: ${theme.$.color.overlay.background};
    `
    : ''
  }
`
interface Props extends StyledProps {
  breakpoint: Breakpoint | null
  hasBackground: boolean
}

export default styled(Div)<Props>(({ theme, breakpoint, hasBackground }: Props) => `
  position: fixed;
  z-index: ${theme.$.zIndex.overlay};

  ${breakpoint
    ? `
      @media(max-width: ${theme.$.length.breakpoint[breakpoint] - 1}px) {
        ${overlayCSS({ theme, hasBackground })};
      }
    `
    : overlayCSS({ theme, hasBackground })
  }
`)
