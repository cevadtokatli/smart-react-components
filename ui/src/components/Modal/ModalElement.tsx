import Div from '@smart-react-components/core/Element/Div'
import Svg from '@smart-react-components/core/Element/Svg'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import styled from 'styled-components'
import ModalHeader from '../../Modal/ModalHeader'
import ModalTitle from '../../Modal/ModalTitle'

interface Props extends
  StyledProps,
  ResponsiveProp<'modalSize', SizeProp> {
  hasBorder: boolean
  isBlock: boolean
  isCentered: boolean
  isFullScreen: boolean
  isStretched: boolean
  shape: ShapeProp
}

export default styled(Div).attrs<Props>(({ isBlock, isFullScreen }: Props) => ({
  getModalSize: (v, t) => `
    ${(!isBlock && !isFullScreen)
      ? `
        width: ${t.$.size.modal[v].width};
      `
      : ''
    }

    > * {
      padding: ${t.$.size.modal[v].padding.y} ${t.$.size.modal[v].padding.x};
    }

    ${ModalHeader} {
      ${ModalTitle} {
        font-size: ${t.$.size.modal[v].titleFontSize};
      }

      ${Svg} {
        margin: -${t.$.size.modal[v].padding.y} -${t.$.size.modal[v].padding.x} -${t.$.size.modal[v].padding.y} 0;
        padding: ${t.$.size.modal[v].padding.y} ${t.$.size.modal[v].padding.x};
        height: ${t.$.size.icon[v]};
        width: ${t.$.size.icon[v]};
      }
    }
  `,
}))<Props>(({ theme, hasBorder, isBlock, isCentered, isFullScreen, isStretched, shape }: Props) => `
  background: ${theme.$.color.dynamic.background};
  box-sizing: border-box;
  color: ${theme.$.color.dynamic.bodyFont};
  display: flex;
  fill: currentcolor;
  flex-direction: column;
  max-height: 100%;

  ${(isBlock || isFullScreen)
    ? `
      width: 100%;
    `
    : ''
  }

  ${(isStretched || isFullScreen)
    ? `
      align-self: stretch;  
    `
    : `
      align-self: ${isCentered ? 'center' : 'flex-start'};
    `
  }

  ${!isFullScreen
    ? `
      border-radius: ${theme.$.radius.modal[shape]};

      ${hasBorder
        ? `
          border: solid 1px ${theme.$.color.dynamic.accent};
        `
        : ''
      }

      > *:first-child {
        border-top-left-radius: ${theme.$.radius.modal[shape]};
        border-top-right-radius: ${theme.$.radius.modal[shape]};
      }

      > *:last-child {
        border-bottom-left-radius: ${theme.$.radius.modal[shape]};
        border-bottom-right-radius: ${theme.$.radius.modal[shape]};
      }
    `
    : ''
  }
`)
