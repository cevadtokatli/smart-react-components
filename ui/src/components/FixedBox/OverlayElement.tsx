import styled from 'styled-components'
import OverlayElement, { Props as OverlayElementProps } from '../Overlay/OverlayElement'

interface Props extends OverlayElementProps {
  duration: number
}

const overlayAnimation = duration => `
  will-change: opacity;

  &.src-fixed-box-show {
    opacity: 0;
  }

  &.src-fixed-box-show-active {
    opacity: 1;
  }

  &.src-fixed-box-hide {
    opacity: 1;
  }

  &.src-fixed-box-hide-active {
    opacity: 0;
  }

  &.src-fixed-box-show-active,
  &.src-fixed-box-hide-active {
    transition: opacity ${duration}ms 0s ease-in-out;
  }
`

const fixedBoxAnimation = duration => `
  > * {
    will-change: opacity, transform;
  }

  &.src-fixed-box-show > * {
    opacity: 0;
    transform: translateY(60px);
  }

  &.src-fixed-box-show-active > * {
    opacity: 1;
    transform: translateY(0);
  }

  &.src-fixed-box-hide > * {
    opacity: 1;
    transform: translateY(0);
  }

  &.src-fixed-box-hide-active > * {
    opacity: 0;
    transform: translateY(60px);
  }

  &.src-fixed-box-show-active,
  &.src-fixed-box-hide-active {
    > * {
      transition: ${duration}ms 0s ease-in-out;
      transition-property: opacity, transform;
    }
  }
`

export default styled(OverlayElement)<Props>(({ theme, breakpoint, duration }: Props) => `
  ${breakpoint
    ? `
      @media(max-width: ${theme.$.length.breakpoint.medium - 1}px) {
        align-items: center;

        ${overlayAnimation(duration)}
      }

      @media(min-width: ${theme.$.length.breakpoint.medium}px) {
        ${fixedBoxAnimation(duration)}
      }
    `
    : `
      ${fixedBoxAnimation(duration)}
    `
  }
`)
