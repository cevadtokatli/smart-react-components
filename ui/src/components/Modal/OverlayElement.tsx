import styled from 'styled-components'
import ModalElement from './ModalElement'
import OverlayElement, { Props as OverlayElementProps } from '../Overlay/OverlayElement'

interface Props extends OverlayElementProps {
  duration: number
  isFullScreen: boolean
}

export default styled(OverlayElement)<Props>(({ theme, duration, isFullScreen }: Props) => `
  ${isFullScreen
    ? `
      padding: 0;  
    `
    : ''
  }

  will-change: opacity;

  > ${ModalElement} {
    will-change: transform;
  }

  &.src-modal-show {
    opacity: 0;

    > ${ModalElement} {
      transform: translateY(-${theme.$.length.overlay.space.y});
    }
  }

  &.src-modal-show-active {
    opacity: 1;

    > ${ModalElement} {
      transform: translateY(0);
    }
  }

  &.src-modal-hide {
    opacity: 1;

    > ${ModalElement} {
      transform: translateY(0);
    }
  }

  &.src-modal-hide-active {
    opacity: 0;

    > ${ModalElement} {
      transform: translateY(-${theme.$.length.overlay.space.y});
    }
  }

  &.src-modal-show-active,
  &.src-modal-hide-active {
    transition: opacity ${duration}ms 0s ease-in-out;

    > ${ModalElement} {
      transition: transform ${duration}ms 0s ease-in-out;
    }
  }
`)
