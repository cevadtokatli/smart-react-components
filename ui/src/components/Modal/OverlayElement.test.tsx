import { wrapTheme } from '@smart-react-components/core/test'
import { render } from '@testing-library/react'
import React from 'react'
import ModalElement from './ModalElement'
import OverlayElement from './OverlayElement'

describe('<OverlayElement />', () => {
  let props
  let modalProps

  beforeAll(() => {
    props = {
      duration: 300,
      hasBackground: true,
      isDisplayedWhenBreakpointNull: true,
      isFullScreen: false,
    }

    modalProps = {
      isBlock: false,
      isCentered: false,
      isFullScreen: false,
      isStretched: false,
      modalSize: 'medium',
      shape: 'rectangle',
    }
  })

  it('should render component', () => {
    const { asFragment } = render(wrapTheme(<OverlayElement {...props}><ModalElement {...modalProps} /></OverlayElement>))
    expect(asFragment()).toMatchSnapshot()
  })

    it('should be full screen', () => {
    const { asFragment } = render(wrapTheme(<OverlayElement {...props} isFullScreen><ModalElement {...modalProps} isFullScreen /></OverlayElement>))
    expect(asFragment()).toMatchSnapshot()
  })
})
