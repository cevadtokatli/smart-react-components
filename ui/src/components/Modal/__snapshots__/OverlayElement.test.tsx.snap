// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OverlayElement /> should be full screen 1`] = `
<DocumentFragment>
  .c3 > * {
  padding: 1rem 1rem;
}

.c4 {
  background: #fff;
  box-sizing: border-box;
  color: #323a46;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  fill: currentcolor;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 100%;
  width: 100%;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

.c0 {
  position: fixed;
  z-index: 2000;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  left: 0;
  padding: 1.75rem;
  top: 0;
  touch-action: none;
  width: 100%;
  background: rgba(0,0,0,.3);
}

.c1 {
  padding: 0;
  will-change: opacity;
}

.c1 > .c2 {
  will-change: transform;
}

.c1.src-modal-show {
  opacity: 0;
}

.c1.src-modal-show > .c2 {
  -webkit-transform: translateY(-1.75rem);
  -ms-transform: translateY(-1.75rem);
  transform: translateY(-1.75rem);
}

.c1.src-modal-show-active {
  opacity: 1;
}

.c1.src-modal-show-active > .c2 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c1.src-modal-hide {
  opacity: 1;
}

.c1.src-modal-hide > .c2 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c1.src-modal-hide-active {
  opacity: 0;
}

.c1.src-modal-hide-active > .c2 {
  -webkit-transform: translateY(-1.75rem);
  -ms-transform: translateY(-1.75rem);
  transform: translateY(-1.75rem);
}

.c1.src-modal-show-active,
.c1.src-modal-hide-active {
  -webkit-transition: opacity 300ms 0s ease-in-out;
  transition: opacity 300ms 0s ease-in-out;
}

.c1.src-modal-show-active > .c2,
.c1.src-modal-hide-active > .c2 {
  -webkit-transition: -webkit-transform 300ms 0s ease-in-out;
  -webkit-transition: transform 300ms 0s ease-in-out;
  transition: transform 300ms 0s ease-in-out;
}

<div
    class=" c0 c1"
  >
    <div
      class="c2 c3 c4"
      shape="rectangle"
    />
  </div>
</DocumentFragment>
`;

exports[`<OverlayElement /> should render component 1`] = `
<DocumentFragment>
  .c3 {
  width: 500px;
}

.c3 > * {
  padding: 1rem 1rem;
}

.c4 {
  background: #fff;
  box-sizing: border-box;
  color: #323a46;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  fill: currentcolor;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 100%;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  border-radius: .3rem;
}

.c4 > *:first-child {
  border-top-left-radius: .3rem;
  border-top-right-radius: .3rem;
}

.c4 > *:last-child {
  border-bottom-left-radius: .3rem;
  border-bottom-right-radius: .3rem;
}

.c0 {
  position: fixed;
  z-index: 2000;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  left: 0;
  padding: 1.75rem;
  top: 0;
  touch-action: none;
  width: 100%;
  background: rgba(0,0,0,.3);
}

.c1 {
  will-change: opacity;
}

.c1 > .c2 {
  will-change: transform;
}

.c1.src-modal-show {
  opacity: 0;
}

.c1.src-modal-show > .c2 {
  -webkit-transform: translateY(-1.75rem);
  -ms-transform: translateY(-1.75rem);
  transform: translateY(-1.75rem);
}

.c1.src-modal-show-active {
  opacity: 1;
}

.c1.src-modal-show-active > .c2 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c1.src-modal-hide {
  opacity: 1;
}

.c1.src-modal-hide > .c2 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c1.src-modal-hide-active {
  opacity: 0;
}

.c1.src-modal-hide-active > .c2 {
  -webkit-transform: translateY(-1.75rem);
  -ms-transform: translateY(-1.75rem);
  transform: translateY(-1.75rem);
}

.c1.src-modal-show-active,
.c1.src-modal-hide-active {
  -webkit-transition: opacity 300ms 0s ease-in-out;
  transition: opacity 300ms 0s ease-in-out;
}

.c1.src-modal-show-active > .c2,
.c1.src-modal-hide-active > .c2 {
  -webkit-transition: -webkit-transform 300ms 0s ease-in-out;
  -webkit-transition: transform 300ms 0s ease-in-out;
  transition: transform 300ms 0s ease-in-out;
}

<div
    class=" c0 c1"
  >
    <div
      class="c2 c3 c4"
      shape="rectangle"
    />
  </div>
</DocumentFragment>
`;
